src/Crate/PDO/PDO.php

```diff
@@ -300,75 +338,98 @@ public function setAttribute($attribute, $value)
 
     /**
      * {@inheritDoc}
+     *
+     * @throws \Crate\PDO\Exception\PDOException
      */
     public function getAttribute($attribute)
     {
         switch ($attribute) {
-            case PDO::ATTR_PERSISTENT:
+            case self::ATTR_PERSISTENT:
                 return false;
 
-            case PDO::ATTR_PREFETCH:
+            case self::ATTR_PREFETCH:
                 return false;
 
-            case PDO::ATTR_CLIENT_VERSION:
+            case self::ATTR_CLIENT_VERSION:
                 return self::VERSION;
 
-            case PDO::ATTR_SERVER_VERSION:
-                return $this->client->getServerVersion();
+            case self::ATTR_SERVER_VERSION:
+                return $this->serverPool->getServerVersion();
 
-            case PDO::ATTR_SERVER_INFO:
-                return $this->client->getServerInfo();
+            case self::ATTR_SERVER_INFO:
+                return $this->serverPool->getServerInfo();
 
-            case PDO::ATTR_TIMEOUT:
+            case self::ATTR_TIMEOUT:
                 return $this->attributes['timeout'];
```

Comment:
`$attributes` looks like a nightmare. Isn't it maybe better to define single properties? That would be more efficient and easier to introspect

Reply:
They are fine, and extremely effective in PHP 7.2, but they are terrible to read. In this case, I must admit that they are indeed non-homogeneous.